deploy-techdocs/action.yaml

```diff
@@ -16,6 +16,9 @@ inputs:
   AWS_TECHDOCS_REGION:
     description: 'aws region'
     required: true
+  working-directory:
+    description: Working directory of where to run the command.
+    required: false
```

Comment:
Would a default of "." be sensible here as to not break any dependent workflows that have not been updated to pass this value?

Reply:
@lukewhiting The default for working directory for a `run` step is the current, so this would change anything. But, I am happy to add `.` as a default, to make it more explicit. 🙂

Reply:
OK. I'm just wondering that happens if that input is not passed by the calling workflow. IE a downstream flow hasn't been updated to pass this new input and it gets templated in here to `working-directory: ""`. Would that cause an error?

Perhaps it's fine and defaults to a nop if it's set to `""` and perhaps `.` is not right but `${{ github.workspace }}` or something. Would like to know the behaviour before we merge (You can test by repointing one of the upstream workflows to your working branch instead of main)

Reply:
Fair enough. If you need proof. I'll see what I can do tomorrow 😄

Reply:
@lukewhiting I have tested the action with both no and an empty `working-directory` input. And it's using the repo root in both cases.

<br>

### No `working-directory` input

```yaml
- name: deploy-techdocs - no working-directory input
  uses: https://github.com/josa42/bw-actions/tree/josa42/allow-setting-a-working-directory/deploy-techdocsjosa42/allow-setting-a-working-directory
  with:
    entity: consumer-research
    TECHDOCS_S3_BUCKET_NAME: ***
    AWS_TECHDOCS_ACCESS_KEY_ID: ***
    AWS_TECHDOCS_SECRET_ACCESS_KEY: ***
    AWS_TECHDOCS_REGION: ***
```

<img width="804" alt="image" src="https://user-images.githubusercontent.com/423234/177701775-4fe63811-58c6-4c53-bf87-36e1f3948d26.png">

<br>

### Empty string `working-directory` input

```yaml
- name: deploy-techdocs - no working-directory input
  uses: https://github.com/josa42/bw-actions/tree/josa42/allow-setting-a-working-directory/deploy-techdocsjosa42/allow-setting-a-working-directory
  with:
    entity: consumer-research
    working-directory: ''
    TECHDOCS_S3_BUCKET_NAME: ***
    AWS_TECHDOCS_ACCESS_KEY_ID: ***
    AWS_TECHDOCS_SECRET_ACCESS_KEY: ***
    AWS_TECHDOCS_REGION: ***
```

<img width="745" alt="image" src="https://user-images.githubusercontent.com/423234/177701855-82abc0c4-5422-4045-9227-56eb3059680c.png">


See workflow run in consumer-research: https://github.com/BrandwatchLtd/consumer-research/runs/7227672577?check_suite_focus=true

Reply:
Perfect :D